packages/expect/src/jest-utils.ts

```diff
@@ -159,6 +159,15 @@ function eq(
     // RegExps are compared by their source patterns and flags.
     case '[object RegExp]':
       return a.source === b.source && a.flags === b.flags
+    case '[object Temporal.Instant]':
+    case '[object Temporal.ZonedDateTime]':
+    case '[object Temporal.PlainDateTime]':
+    case '[object Temporal.PlainDate]':
+    case '[object Temporal.PlainTime]':
+    case '[object Temporal.PlainYearMonth]':
+    case '[object Temporal.PlainMonthDay]':
+    case '[object Temporal.Duration]':
+      return a.toString() === b.toString()
```

Comment:
Do you know which is preferred whether `toString` equality and builtin `equals` method? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/ZonedDateTime/equals

It looks like there are several old issues mentions calendar comparison (such as https://github.com/tc39/proposal-temporal/issues/625), but the discussion seem quite technical :sweat_smile:

Reply:
`.equals()` seems to compare including the calendar, so I think it's a good suggestion. Let's do it! 

For `Temporal.Duration` this `.equals()` method doesn't exist, but I wasn't sure about that one to begin with. For Durations, there IS actually ambiguity. For example, are 60 seconds and 1 minute the same? If you compare the duration in time, yes; but if you literally compare all their components, no.

My suggestion is to leave out Duration for now.

Reply:
Indeed it looks like Duration is tricky. https://github.com/tc39/proposal-temporal/issues/856 seems to explain the details.

As a builtin equality, I think choosing `toString` comparison makes sense since that's the most strict way to do it. Users can still add own custom equality tester to loosen it if desired.